src/nevermined/Token.ts

```diff
@@ -42,7 +42,7 @@ export class TokenUtils extends Instantiable {
   /**
    * Get token total supply
    */
-  public async getTotalSupply(): Promise<number> {
+  public async getTotalSupply(): Promise<BigNumber> {
     return await this.nevermined.keeper.token.totalSupply()
```

Comment:
This class is problematic.

In the keeper the token is optional and is only loaded in networks where we have the nevermined token https://github.com/nevermined-io/sdk-js/blob/6690c6cc9ab1c232812d0c49bc34a796125dda49/src/keeper/Keeper.ts#L164-L165

For networks without nevermined token `token` is undefined so calling any of these methods will fail. Here is a fail point when registering https://github.com/nevermined-io/sdk-js/blob/6690c6cc9ab1c232812d0c49bc34a796125dda49/src/nevermined/api/RegistryBaseApi.ts#L69

When using custom erc20 tokens we need to load them and call methods directly on the token

```ts
customErc20Token = await nevermined.contracts.loadErc20(erc20TokenAddress)
const tokenDecimals = await customErc20Token.decimals()
```
anything else will fail including `nevermined.keeper.token` and `nevermined.utils.token`

We could simply inject it into the sdk like you did in the cli but I think this can lead to problems, for instance it only works correctly assuming that we are always using the same instance on the sdk. Also don't know how this will behave when we want to support multiple tokens.

I think the problem that is happening on mainnet is that assets don't have a price and it defaults to `new AssetPrice()` so  `AssetPrice.getTokenAddress()` will return undefined and the sdk will try to use the nevermined token by default.

One way to fix this for now would be to make sure we always set the price in the NFTAttributes even if it's zero. something like:

```
const price = new AssetPrice()
price.setTokenAddress(...)

Reply:
good point, probably is better to move the initialization of the token address to the default AssetAttributes